backend/src/components/edx/exchange.js

```diff
@@ -252,15 +252,19 @@ async function markAsClosed(req, res) {
   ])
     .then(async ([statusCodeResponse, ministryTeamCodeResponse, dataResponse]) => {
       if (statusCodeResponse && ministryTeamCodeResponse && dataResponse) {
+        const userInfo = utils.getUser(req);
+        let userRoles = userInfo.realm_access.roles;
         let roleCheckPass = false;
         let ownerTeamCheckPass = false;
         let exchange = dataResponse;
         if(exchange['ministryOwnershipTeamID']){
           let matchedMinTeam = ministryTeamCodeResponse.find(minstryTeam => minstryTeam['ministryOwnershipTeamId'] === exchange['ministryOwnershipTeamID']);
           if(matchedMinTeam['groupRoleIdentifier']) {
```

Comment:
There's a missing condition here; we need to check if the user has the correct team role. Check if the `userRoles` object contains the `groupRoleIdentifier` you found. You can use a lambda function for this if you want.

Reply:
`userRoles` is just an array of roles. The userInfo object does not contain any team related data. I could call the getEdxUsers api, filter out the current user and use edxUserSchools[]->edxUserSchoolRoles[]->edxRoleCode. The other option I see is to use the groupRoleIdentifier field from the ministryTeamCodeResponse. What do you think?